projects/plugins/jetpack/_inc/client/traffic/seo.jsx

```diff
@@ -2,36 +2,116 @@
+import { __, _x, _n, sprintf } from '@wordpress/i18n';
+import { FacebookPreview, TwitterPreview, SearchPreview } from '@automattic/social-previews';
+import SocialLogo from 'social-logos';
 
 /**
  * Internal dependencies
  */
 import analytics from 'lib/analytics';
-import Card from 'components/card';
 import getRedirectUrl from 'lib/jp-redirect';
 import { withModuleSettingsFormHelpers } from 'components/module-settings/with-module-settings-form-helpers';
 import SettingsCard from 'components/settings-card';
 import SettingsGroup from 'components/settings-group';
 import { ModuleToggle } from 'components/module-toggle';
 import ConnectUserBar from 'components/connect-user-bar';
+import { FormLabel, FormTextarea, FormFieldset } from 'components/forms';
+import FoldableCard from 'components/foldable-card';
+import CustomSeoTitles from './seo/custom-seo-titles.jsx';
+import SimpleNotice from 'components/notice';
+
+export const conflictingSeoPlugins = [
+	'Yoast SEO',
+	'Yoast SEO Premium',
+	'All In One SEO Pack',
+	'All in One SEO Pack Pro',
+];
+
+// Returns first conflicting plugin
+export const getConflictingSeoPlugin = plugins => {
+	const conflictingPlugins = Object.keys( plugins ).filter( plugin => {
+		return plugins[ plugin ].active && conflictingSeoPlugins.includes( plugins[ plugin ].Name );
+	} );
+
+	return plugins[ conflictingPlugins[ 0 ] ];
+};
 
 export const SEO = withModuleSettingsFormHelpers(
 	class extends Component {
+		constants = {
+			frontPageMetaMaxLength: 300,
+			frontPageMetaSuggestedLength: 159,
+			moduleOptionsArray: [ 'advanced_seo_front_page_description', 'advanced_seo_title_formats' ],
+			siteIconPreviewSize: 512,
+		};
+
 		trackConfigureClick = () => {
 			analytics.tracks.recordJetpackClick( 'configure-seo' );
 		};
 
+		SocialPreviewGoogle = siteData => (
+			<SearchPreview
+				title={ siteData.title }
+				url={ siteData.url }
+				description={ siteData.frontPageMetaDescription }
+			/>
+		);
+
+		SocialPreviewFacebook = siteData => (
+			<FacebookPreview
+				title={ siteData.title }
+				url={ siteData.url }
+				type="website"
+				description={ siteData.frontPageMetaDescription }
+				image={ siteData.image }
+			/>
+		);
+
+		SocialPreviewTwitter = siteData => (
+			<TwitterPreview
+				title={ siteData.title }
+				url={ siteData.url }
+				type="summary"
+				description={ siteData.frontPageMetaDescription }
+				image={ siteData.image }
+			/>
+		);
+
+		updateCustomSeoTitleInputState = newCustomSeoTitles => {
+			this.props.updateFormStateOptionValue( 'advanced_seo_title_formats', newCustomSeoTitles );
+		};
+
 		render() {
-			const isActive = this.props.getOptionValue( 'seo-tools' );
+			const seo = this.props.getModule( 'seo-tools' );
+			const isSeoActive = this.props.getOptionValue( seo.module );
+			const isFetchingPluginsData = this.props.pluginsData.requests.isFetchingPluginsData;
+			const hasConflictingSeoPlugin = getConflictingSeoPlugin( this.props.pluginsData.items );
```

Comment:
Would it make sense to import `fetchPluginsData`, `isFetchingPluginsData`, and `isPluginActive` from `state/site/plugins` right here, and rely on `isPluginActive` to check for each conflicting plugin?

Reply:
Nice, didn't realize we had those functions: https://github.com/Automattic/jetpack/pull/18810/commits/2a06b19e6d6c96b16f1b3eb89d2cd31c6f409f2e